main/test/ca/uwaterloo/flix/library/TestSet.flix

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright 2016 Liam Palmer
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/////////////////////////////////////////////////////////////////////////////
+// size                                                                    //
+/////////////////////////////////////////////////////////////////////////////
+@test
+def size01: Bool = Set/size(#{}) `assertEq!` 0
+
+@test
+def size02: Bool = Set/size(#{1}) `assertEq!` 1
+
+/////////////////////////////////////////////////////////////////////////////
+// empty                                                                   //
+/////////////////////////////////////////////////////////////////////////////
+@test
+def empty01: Bool = Set/empty() `assertEq!` #{}
+
+/////////////////////////////////////////////////////////////////////////////
+// singleton                                                               //
+/////////////////////////////////////////////////////////////////////////////
+@test
+def singleton01: Bool = Set/singleton(1) `assertEq!` #{1}
+
+@test
+def singleton02: Bool = Set/singleton(2) `assertEq!` #{2}
+
+@test
+def singleton03: Bool = Set/singleton(true) `assertEq!` #{true}
+
+/////////////////////////////////////////////////////////////////////////////
+// insert                                                                  //
+/////////////////////////////////////////////////////////////////////////////
+@test
+def insert01: Bool = Set/insert(1, #{}) `assertEq!` #{1}
+
+@test
+def insert02: Bool = Set/insert(1, #{1}) `assertEq!` #{1}
+/*
+@test
+def insert03: Bool = Set/insert(1, #{2}) `assertEq!` #{1, 2}
```

Comment:
Follow up:

On circleci is see the following error:

```
java.lang.RuntimeException: Attempt to retrieve value from Failure. The errors are: UndefinedRef(List/length,Set,SourceLocation(Internal(Set.flix,/*
```

which suggests that List is not included.

That is either because you did not include it explicitely here:

https://github.com/flix/flix/blob/master/main/test/ca/uwaterloo/flix/util/FlixTest.scala#L30

or implicitely here:

https://github.com/flix/flix/blob/master/main/src/ca/uwaterloo/flix/api/Flix.scala#L53

Ultimately we want the entire standard library to be included by default, but I don't know if you can include List in Flix.scala without breaking other tests.